### PR TITLE
Avoid duplicate values in MapperTestCase#testFetchMany

### LINKED ISSUE #71053: [CI] VersionStringFieldMapperTests#testFetchMany failed
**Build scan**:

https://gradle-enterprise.elastic.co/s/7mri5usps5fui

**Repro line**:

./gradlew ':x-pack:plugin:mapper-version:test' --tests "org.elasticsearch.xpack.versionfield.VersionStringFieldMapperTests.testFetchMany" -Dtests.seed=257DB53B6581EE8F -Dtests.security.manager=true -Dtests.locale=es-PY -Dtests.timezone=Asia/Calcutta -Druntime.java=11

**Reproduces locally?**:

Yes 

**Applicable branches**:

master

**Failure history**:

This started failing on March 26. on 7.x, four failures to far so seems only to happen with certain random seeds.

**Failure excerpt**:

```
java.lang.AssertionError: fetching [90, 64alpha12.9.72, 31, 45.37, 31, 47.78]
    Expected: iterable with items ["31", "45.37", "47.78", "90", "64alpha12.9.72"] in any order
         but: not matched: "31"
        at __randomizedtesting.SeedInfo.seed([257DB53B6581EE8F:9EB254E5FE002DE7]:0)
        at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
        at org.junit.Assert.assertThat(Assert.java:956)
        at org.elasticsearch.index.mapper.MapperTestCase.lambda$assertFetch$19(MapperTestCase.java:577)
        at org.elasticsearch.index.mapper.MapperServiceTestCase.withLuceneIndex(MapperServiceTestCase.java:223)
        at org.elasticsearch.index.mapper.MapperTestCase.assertFetch(MapperTestCase.java:551)
        at org.elasticsearch.index.mapper.MapperTestCase.testFetchMany(MapperTestCase.java:496)
```

### PR DESCRIPTION
**Pull request by cbuescher**
The test currently generates a list of random values and checks whether
retrieval of these values via doc values is equivallent to fetching them with a
value fetcher from source. If the random value array contains a duplicate value,
we will only get one back via doc values, but fetching from source will return
both, which is a case we should probably avoid in this test.

Closes #71053

### CODE REVIEW

File: test/framework/src/main/java/org/elasticsearch/index/mapper/MapperTestCase.java [hunk 1]
```diff
@@ -491,7 +491,8 @@ public final void testFetchMany() throws IOException {
             int count = between(2, 10);
             List<Object> values = new ArrayList<>(count);
             while (values.size() < count) {
-                values.add(generateRandomInputValue(ft));
+                // values should be unique since we also get de-duplication fetching from docvalues
+                values.add(randomValueOtherThanMany(v -> values.contains(v), () -> generateRandomInputValue(ft)));
```
Comment (cbuescher) [inline test/framework/src/main/java/org/elasticsearch/index/mapper/MapperTestCase.java:495]: I'm not entirely sure if it is better to fix this in the setup or in the assertion in the end. We might also de-dup the results from the native fetchers before comparing, wdyt?

**Comment by nik9000** [comment]
Im not in a good place to double check, but I think duplicate generated
values is ok because we dedupe on fetch. Or we can if we need to. I forget
which.

**Comment by dnhatn** [comment]
@cbuescher I've muted this test in 1edc7c68490d68e32c02c999a18468f01a9f566c.

**Comment by cbuescher** [comment]
@elasticmachine update branch

**Comment by cbuescher** [comment]
@elasticmachine run elasticsearch-ci/eql-correctness

**Comment by cbuescher** [comment]
@elasticmachine run elasticsearch-ci/1

**Review by nik9000 — COMMENTED** [review]
I *thought* numbers kept duplicates. I *think* its just `keyword` style stuff that removes duplicates. Maybe this is something we should only do on those types?

Also - would it be cleaner to remove the duplicates after fetching rather than before indexing? It feels like cheating to ask for `5` values and only get back `1` because of duplicates.

**Comment by cbuescher** [comment]
>  I think its just keyword style stuff that removes duplicates

This is interesting. I ran some checks by forcing duplicates in the generated random values and found several where the `testFetchMany` test fails then. I moved the dedup code closer to the assertion like you suggested, we should remove duplicates from what we fetch via 'fields' before comparing it to doc values in this case. I think having the duplicates in the `fields` output is right in these cases though. So this is ready for another look if you like.

**Review by nik9000 — APPROVED** [review]
LGTM

**Pull request merged**